### PR TITLE
docs: Use `current.yaml` for current changelog

### PR DESCRIPTION
**Pull request by phlax**
- use upstreamed `Project` class
- use `current.yaml` for current changelog file

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/docs/versions_check.py [hunk 1]
```diff
@@ -103,25 +104,20 @@ class VersionsChecker(checker.Checker):
     checks = ("changelogs", "pending", "version")
 
     @property
-    def changelogs(self) -> Tuple[pathlib.Path, ...]:
-        return tuple(pathlib.Path(p) for p in self.args.changelogs)
+    def changelog_sections(self) -> pathlib.Path:
```
Comment (phlax) [inline tools/docs/versions_check.py:107]: remove

File: bazel/repositories.bzl [hunk 2]
```diff
@@ -94,7 +94,7 @@ py_library(name = "envoy_repo", srcs = ["__init__.py"], visibility = ["//visibil
 _envoy_repo = repository_rule(
     implementation = _envoy_repo_impl,
     attrs = {
-        "envoy_root": attr.label(default = "@envoy//:BUILD"),
+        "envoy_version": attr.label(default = "@envoy//:VERSION.txt"),
```
Comment (phlax) [inline bazel/repositories.bzl:97]: this change was necessary, otherwise changing the version doesnt expire the cache and the version is set incorrectly

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks LGTM. Can you merge main and also s/current.rst/current.yaml everywhere per @alyssawilk? 

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**